templates/views/movie_profile.pug

```diff
@@ -38,10 +42,41 @@ block content
         +sectionParagraph(movie.summary.html)    
       
     //- video section
-    if movie.video
+    if movie.video && movie.video.includes("mediacentral.ucl.ac.uk")
       section#video.pv3.flex.flex-column
+        - const videoLink = movie.video
+        - const videoId = movie.video.split('/').pop();
+        - const baseUrl = "https://mediacentral.ucl.ac.uk/player?autostart=n&fullscreen=y&width=0&height=0&quality=hd&captions=n&chapterId=0"
+        - const iframeUrl = baseUrl + "&videoId=" + videoId
         +sectionHeading("Video")
-        #video-container.h5.bg-gray.w-100.mw7.self-center
-      
-     
-                
+        div.relative.pb-video.pt2.h0
+          iframe.absolute.top0.left0.w-100.h-100(allowfullscreen src=iframeUrl frameborder="0" scrolling="no")
+        if movie.videoLinks.html
+          +subHeading("More video links")
+          #video-links.pa2
+            !=movie.videoLinks.html
+          
+    //- gallery section
+    if movie.gallery
+      section#gallery.pv3.flex.flex-column
+        +sectionHeading("Image gallery")
+        div.slider-holder.w-90.h4.h6-ns.bg-yellow.center.mt0.overflow-hidden.tc
+          - let index = 1
+          each img in movie.gallery  
```

Comment:
Okay, so you get the index for free

https://pugjs.org/language/iteration.html